src/framing/Frame.cpp

```diff
@@ -95,78 +95,120 @@ std::ostream& operator<<(std::ostream& os, ErrorCode errorCode) {
   return os << "ErrorCode(" << static_cast<uint32_t>(errorCode) << ")";
 }
 
-std::ostream& operator<<(std::ostream& os, FrameFlags frameFlags) {
-  // TODO Match the Flag names with the AllowedFlags in the Frame declarations
+constexpr const char sFrameFlags_EMPTY[] = "0x00";
+constexpr const char sFrameFlags_METADATA[] = "METADATA";
+constexpr const char sFrameFlags_RESUME_ENABLE[] = "RESUME_ENABLE";
+constexpr const char sFrameFlags_LEASE[] = "LEASE";
+constexpr const char sFrameFlags_KEEPALIVE_RESPOND[] = "KEEPALIVE_RESPOND";
+constexpr const char sFrameFlags_FOLLOWS[] = "FOLLOWS";
+constexpr const char sFrameFlags_COMPLETE[] = "COMPLETE";
+constexpr const char sFrameFlags_NEXT[] = "NEXT";
+
+std::string to_string(
+    FrameFlags frameFlags,
+    const std::vector<std::pair<FrameFlags, std::string>>& allowedFlags) {
+  FrameFlags foundFlags = FrameFlags::EMPTY;
 
   std::stringstream ss;
   std::string delimeter = "";
-  if (!!(frameFlags & FrameFlags::NEXT)) {
-    ss << "NEXT";
-    delimeter = "|";
-  }
-  if (!!(frameFlags & FrameFlags::COMPLETE)) {
-    ss << delimeter << "COMPLETE";
-    delimeter = "|";
-  }
-  if (!!(frameFlags & FrameFlags::FOLLOWS)) {
-    ss << delimeter << "FOLLOWS";
-    delimeter = "|";
+  for (auto& pair : allowedFlags) {
+    if (!!(frameFlags & pair.first)) {
+      ss << delimeter << pair.second;
+      delimeter = "|";
+      foundFlags |= pair.first;
+    }
   }
-  if (!!(frameFlags & FrameFlags::METADATA)) {
-    ss << delimeter << "METADATA";
-    delimeter = "|";
-  }
-  if (!!(frameFlags & FrameFlags::IGNORE)) {
-    ss << delimeter << "IGNORE";
-    delimeter = "|";
-  }
-  if (!delimeter.empty()) {
-    return os << ss.str();
+
+  if (foundFlags != frameFlags) {
+    ss << frameFlags;
+  } else if (delimeter.empty()) {
+    ss << sFrameFlags_EMPTY;
   }
-  return os << "EMPTY";
+  return ss.str();
+}
+
+std::ostream& operator<<(std::ostream& os, FrameFlags frameFlags) {
+  std::bitset<16> flags(static_cast<uint16_t>(frameFlags));
+  return os << flags;
 }
 
 std::ostream& operator<<(std::ostream& os, const FrameHeader& header) {
   return os << header.type_ << "[" << header.flags_ << ", " << header.streamId_
             << "]";
 }
+
+std::ostream& operator<<(
+    std::ostream& os,
+    const std::pair<
+        const FrameHeader&,
+        const std::vector<std::pair<FrameFlags, std::string>>&>&
```

Comment:
I do not need the FrameFlags information of the Frame class types. Though I need a mapping between each flag type to corresponding string value per Frame class type. I prepared a map for this. I am open for suggestion, though it seemed required such a map to me.